packages/react-popover/src/components/PopoverTrigger/usePopoverTrigger.ts

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import {
+  makeMergeProps,
+  useMergedRefs,
+  useEventCallback,
+  shouldPreventDefaultOnKeyDown,
+} from '@fluentui/react-utilities';
+import { PopoverTriggerProps, PopoverTriggerState } from './PopoverTrigger.types';
+import { usePopoverContext } from '../../popoverContext';
+
+const mergeProps = makeMergeProps<PopoverTriggerState>({});
+
+/**
+ * Create the state required to render PopoverTrigger.
+ *
+ * The returned state can be modified with hooks such as usePopoverTriggerStyles,
+ * before being passed to renderPopoverTrigger.
+ *
+ * @param props - props from this instance of PopoverTrigger
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ */
+export const usePopoverTrigger = (
+  props: PopoverTriggerProps,
+  defaultProps?: PopoverTriggerProps,
+): PopoverTriggerState => {
+  const setOpen = usePopoverContext(context => context.setOpen);
+  const open = usePopoverContext(context => context.open);
+  const triggerRef = usePopoverContext(context => context.triggerRef);
+  const openOnHover = usePopoverContext(context => context.openOnHover);
+  const openOnContext = usePopoverContext(context => context.openOnContext);
+
+  const state = mergeProps(
+    {
+      children: (null as unknown) as React.ReactElement,
+    },
+    defaultProps,
+    props,
+  );
+
+  const onContextMenu = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (openOnContext) {
+      e.preventDefault();
+      setOpen(e, true);
+    }
+    child.props?.onContextMenu?.(e);
+  });
+
+  const onClick = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (!openOnContext) {
+      setOpen(e, !open);
+    }
+    child.props?.onClick?.(e);
+  });
+
+  const onKeyDown = useEventCallback((e: React.KeyboardEvent<HTMLElement>) => {
+    if (shouldPreventDefaultOnKeyDown(e)) {
+      e.preventDefault();
+      (e.target as HTMLElement)?.click();
+    }
+
+    child.props?.onKeyDown?.(e);
+  });
+
+  const onMouseEnter = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (openOnHover && !openOnContext) {
+      setOpen(e, true);
+    }
+    child.props?.onMouseEnter?.(e);
+  });
+
+  const onMouseLeave = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (openOnHover && !openOnContext) {
+      setOpen(e, false);
+    }
+    child.props?.onMouseLeave?.(e);
+  });
+
+  const child = React.Children.only(state.children);
+  state.children = React.cloneElement(child, {
+    'aria-haspopup': 'true',
+    ...child.props,
+
+    // These event handlers must handle original props
```

Comment:
I don't understand the comment, sorry.

Reply:
I will rework it

Reply:
Thanks to some reading, found out that `cloneElement` already does shallow spreading of previous props, so removed this comment and the line above it